CRM/Admin/Form/Setting/UF.php

```diff
@@ -70,26 +70,41 @@ public function buildQuickForm() {
     $this->assign('tablePrefixes', FALSE);
 
     if ($config->userSystem->viewsExists() &&
-      (
-        $config->dsn != $config->userFrameworkDSN || !empty($drupal_prefix)
-      )
-    ) {
-
-      $dsnArray = DB::parseDSN(CRM_Utils_SQL::autoSwitchDSN($config->dsn));
-      $tableNames = CRM_Core_DAO::getTableNames();
-      asort($tableNames);
-      $tablePrefixes = '$databases[\'default\'][\'default\'][\'prefix\']= [';
-      if ($config->userFramework === 'Backdrop') {
-        $tablePrefixes = '$database_prefix = [';
+      ($config->dsn != $config->userFrameworkDSN || !empty($drupal_prefix))) {
+      if (($config->userSystem->is_drupal) && (explode('.', \DRUPAL::VERSION) >= 9)) {
```

Comment:
I agree with the general premise of the PR but this buildQuickForm() function has always been horrible and if we're going to refactor it we really should move parts into the various CRM/Utils/System/XXX.php files instead of "if drupal", "if backdrop", etc.

Technically \DRUPAL::VERSION is a typo since the class name is Drupal but php is weird.

Also note that this may also depend on which version of civicrm_entity is in use. I don't remember off-hand.